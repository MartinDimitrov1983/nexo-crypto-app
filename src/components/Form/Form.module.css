.form {
    display: flex;
    flex-direction: column;
    max-width: 500px;
}
.container {
    display: flex;
    justify-content: space-around;
    align-items: center;
    max-width: 600px;
    gap: 24px;
    margin-bottom: 30px;
    flex-wrap: wrap;
}

@media only screen and (max-width: 560px) {
    .container {
        flex-direction: column;
    }
    .inputs {
        flex-direction: column;
    }
}

.buttonDisabled {
    background: grey;
    cursor: auto;
}

.buttonDisabled:hover {
    scale: 1;
}
