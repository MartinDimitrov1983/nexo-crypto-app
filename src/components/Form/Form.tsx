import React, { ChangeEvent, FormEventHandler } from 'react';
import { Input, Button } from '../';
import { ButttonType } from '../Button/Button';
import { SEARCH_LABEL, CURRENCY_LABEL } from '../../constants';
import styles from './Form.module.css';

export interface FormProps {
    handleSearch: FormEventHandler;
    handleOnChange: Function;
    value: string;
}

const Form = ({ handleSearch, handleOnChange, value }: FormProps) => {
    return (
        <form onSubmit={handleSearch}>
            <div className={styles.container}>
                <Input
                    id="pair"
                    labelName={CURRENCY_LABEL}
                    value={value}
                    handleOnChange={(e: ChangeEvent<HTMLInputElement>) => {
                        e.preventDefault();
                        handleOnChange(e!.target!.value);
                    }}
                />
                <Button
                    type={ButttonType.SUBMIT}
                    shouldShowIcon={true}
                    isDisabled={!value}
                    className={`${!value && styles.buttonDisabled}`}
                >
                    {SEARCH_LABEL}
                </Button>
            </div>
        </form>
    );
};

export default Form;
