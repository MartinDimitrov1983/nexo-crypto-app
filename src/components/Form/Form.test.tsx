import React from 'react';
import { render } from '@testing-library/react';

import Form from './Form';

test('Snapshot Form', () => {
    const { asFragment } = render(
        <Form handleSearch={() => {}} handleOnChange={() => {}} value="test" />,
    );
    expect(asFragment()).toMatchSnapshot();
});
