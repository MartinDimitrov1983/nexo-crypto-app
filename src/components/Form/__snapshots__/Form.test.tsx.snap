// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Form 1`] = `
<DocumentFragment>
  <form>
    <div
      class="container"
    >
      <div
        class="wrapper"
      >
        <label
          class="label"
          for="pair"
        >
          Currency
        </label>
        <input
          class="input"
          id="pair"
          type="text"
          value="test"
        />
      </div>
      <button
        class="button false"
        type="submit"
      >
        <div
          class="content"
        >
          <svg
            fill="white"
            style="margin-right: 5px;"
          >
            search.svg
          </svg>
          <div>
            SEARCH
          </div>
        </div>
      </button>
    </div>
  </form>
</DocumentFragment>
`;
